tests/test_compiler.py

```diff
@@ -367,3 +367,15 @@ def test_source_map(project, compiler):
     source_path = project.contracts_folder / "MultipleDefinitions.sol"
     result = compiler.compile([source_path])[-1]
     assert result.sourcemap.__root__ == "124:87:0:-:0;;;;;;;;;;;;;;;;;;;"
+
+
+def test_add_library(project, account, compiler, connection):
+    with pytest.raises(AttributeError):
+        # Does not exist yet because library is not deployed or known.
+        _ = project.C
+
+    library = project.Set.deploy(sender=account)
+    compiler.add_library(library)
```

Comment:
We _could_ make the plugin aware of all the libraries in the project and then have a method like `add_libraries()` that takes no arguments and loops through the libraries it finds, checks if there any deployments, if so, sets them on the compiler so they are ready to be used....

All that would really save is not needing to specify an arg here though, and it would be cumbersome to implement.